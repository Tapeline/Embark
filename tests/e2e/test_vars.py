from tests.runner import run_embark


def test_variables_in_file(snapshot):
    """Test that variables in file work."""
    proc = run_embark("tests/fixtures/test_vars_in_file.yml")
    assert proc.returncode == 0, (proc.stdout, proc.stderr)
    assert proc.stdout == snapshot


def test_variables_in_separate_file(snapshot):
    """Test that variables in .variables.yml file work."""
    proc = run_embark("tests/fixtures/test_vars_in_sep_file.yml")
    assert proc.returncode == 0, (proc.stdout, proc.stderr)
    assert proc.stdout == snapshot


def test_variables_mixing(snapshot):
    """Test that variables from env, .variables and playbook work together."""
    proc = run_embark(
        "tests/fixtures/test_vars_mixing.yml",
        env={
            "VAR3": "env"
        }
    )
    assert proc.returncode == 0, (proc.stdout, proc.stderr)
    assert proc.stdout == snapshot
