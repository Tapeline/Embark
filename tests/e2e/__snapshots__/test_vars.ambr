# serializer version: 1
# name: test_variables_in_file
  '''
  Hello, World!
  
  '''
# ---
# name: test_variables_in_separate_file
  '''
  variables.yml1
  
  '''
# ---
# name: test_variables_mixing
  '''
  playbook variables.yml2 env
  
  '''
# ---
